altapay.php

```diff
@@ -0,0 +1,2353 @@
+        $cgConf['language'] = $this->context->language->iso_code;
+        $cgConf['uniqueid'] = $cart->id;
+        $cgConf['terminal'] = $terminal->remote_name;
+        $cgConf['cookie'] = isset($_SERVER['HTTP_COOKIE']) ? $_SERVER['HTTP_COOKIE'] : null;
+
+        $callback = array();
+        // callbacks
+        $callback['callback_form'] = $this->context->link->getModuleLink(
+            $this->name,
+            'callbackform',
+            array(),
+            true,
+            $this->context->language->id,
+            $this->context->shop->id
+        );
+        $callback['callback_ok'] = $this->context->link->getModuleLink(
+            $this->name,
+            'callbackok',
+            array(),
+            true,
+            $this->context->language->id,
+            $this->context->shop->id
+        );
+        $callback['callback_fail'] = $this->context->link->getModuleLink(
+            $this->name,
+            'callbackfail',
+            array(),
+            true,
+            $this->context->language->id,
+            $this->context->shop->id
+        );
+        $callback['callback_open'] = $this->context->link->getModuleLink(
+            $this->name,
+            'callbackopen',
+            array(),
+            true,
+            $this->context->language->id,
+            $this->context->shop->id
+        );
+        $callback['callback_notification'] = $this->context->link->getModuleLink(
+            $this->name,
+            'callbacknotification',
+            array(),
+            true,
+            $this->context->language->id,
+            $this->context->shop->id
+        );
+        $callback['callback_redirect'] = $this->context->link->getModuleLink(
+            $this->name,
+            'callbackredirect',
+            array(),
+            true,
+            $this->context->language->id,
+            $this->context->shop->id
+        );
+        $customer = array();
+        // customer info
+        $customer['billing_firstname'] = $this->context->customer->firstname;
+        $customer['billing_lastname'] = $this->context->customer->lastname;
+        $customer['email'] = $this->context->customer->email;
+
+        // billing address
+        $invoice_address = new Address($this->context->cart->id_address_invoice);
+        $country = new Country($invoice_address->id_country);
+        $state = new State($invoice_address->id_state);
+//        $customer['type'] = $userType;$userType
+
+        $customer['billing_address'] = $invoice_address->address1;
+        $customer['billing_city'] = $invoice_address->city;
+        $customer['billing_postal'] = $invoice_address->postcode;
+        $customer['billing_region'] = $state->iso_code;
+        $customer['billing_country'] = $country->iso_code;
+
+        // phone
+        $invoiceAph = $invoice_address->phone;
+        $customer['customer_phone'] = $invoice_address->phone != '' ? $invoiceAph : $invoice_address->phone_mobile;
```

Comment:
Since `$invoiceAph` was set to the content of `$invoice_address->phone` and a non strict comparison is being used, this would be equivilant to:
```suggestion
        $customer['customer_phone'] = $invoiceAph ?: $invoice_address->phone_mobile;
```